src/main/java/bio/terra/service/tabulardata/google/BigQueryPdao.java

```diff
@@ -1592,16 +1651,17 @@ public void validateDeleteRequest(Dataset dataset, List<DataDeletionTableModel>
     }
 
     // we select from the live view here so that the row counts take into account rows that have been hard deleted
-    private static final String rowCountTemplate = "SELECT COUNT(<rowId>) FROM `<project>.<dataset>.<table>`";
+    private static final String rowCountTemplate = "SELECT COUNT(<rowId>) FROM `<datasetProject>.<dataset>.<table>`";
 
     public Map<String, Long> getSnapshotTableRowCounts(Snapshot snapshot) throws InterruptedException {
-        BigQueryProject bigQueryProject = bigQueryProjectForSnapshot(snapshot);
+        //TODO - this needs to happen per source dataset
+        BigQueryProject bigQueryProject = bigQueryProjectForDataset(snapshot.getFirstSnapshotSource().getDataset());
         Map<String, Long> rowCounts = new HashMap<>();
         for (SnapshotTable snapshotTable : snapshot.getTables()) {
             String tableName = snapshotTable.getName();
             String sql = new ST(rowCountTemplate)
                 .add("rowId", PDAO_ROW_ID_COLUMN)
-                .add("project", bigQueryProject.getProjectId())
+                .add("datasetProject", bigQueryProject.getProjectId())
```

Comment:
I _think_ this should select from the snapshot project, no?

Reply:
Ohh - good catch - thanks! I was confused by the next line: `.add("dataset", snapshot.getName())`